src/controller/CHIPDeviceController.cpp

```diff
@@ -1625,6 +1614,14 @@ void DeviceCommissioner::OnNodeIdResolved(const chip::Dnssd::ResolvedNodeData &
                     ChipLogValueX64(nodeData.mPeerId.GetNodeId()));
     VerifyOrReturn(mState == State::Initialized);
 
+    if (mDeviceBeingCommissioned != nullptr && mDeviceBeingCommissioned->GetDeviceId() == nodeData.mPeerId.GetNodeId())
```

Comment:
what if RendezvousCleanup( is called with an error? In that case, nodeid resolved probably will not be called.

What happens if DNSSD is disabled? is that even  possible? why do we have that ifdef and does this mean that no dnssd  means commisioners cannot be implemented?

Reply:
DNSSD should not be ifdef anymore. I don't think we have any alternate mechanism to lookup devices on the operational network. Maybe it's a left over ifdef from when DNSSD/MDNS was initially implemented.

Good catch about `RendezvousCleanup`. I'll take a look at it and update the PR.